homeassistant/components/image_processing/facebox.py

```diff
@@ -45,6 +49,25 @@
 })
 
 
+def check_box_health(url, username, password):
+    """Check the health of the classifier and return its id if healthy."""
+    kwargs = {}
+    if username:
+        kwargs['auth'] = requests.auth.HTTPBasicAuth(username, password)
+    try:
+        response = requests.get(
+            url,
+            **kwargs
+        )
+        if response.status_code == HTTP_UNAUTHORIZED:
+            _LOGGER.error("AuthenticationError on %s", CLASSIFIER)
+            return None
```

Comment:
We should fail setup, ie not add the entity and return from `setup_platform`, if this happens, besides logging the error. If the credentials are bad the platform will never work until the user fixes that.

Reply:
Would you advise that the auth should be checked before instantiaiton of `FaceClassifyEntity` or should the init of `FaceClassifyEntity` fail and return this to setup?

Reply:
Yeah, early is good.